src/clr-angular/forms/combobox/options.ts

```diff
@@ -1,59 +1,140 @@
+
+    <!--Option Groups and Options will be projected here-->
+    <ng-content></ng-content>
+  `,
+  providers: [{ provide: LoadingListener, useExisting: ClrOptions }],
+  host: {
+    '[class.clr-options]': 'true',
+    '[attr.aria-multiselectable]': 'optionSelectionService.multiselectable',
+    '[attr.role]': '"listbox"',
+    '[id]': 'optionsId',
+  },
+})
+export class ClrOptions<T> implements AfterViewInit, LoadingListener, OnDestroy {
+  public loading = false;
+  private subscriptions: Subscription[] = [];
 
   constructor(
-    injector: Injector,
+    public optionSelectionService: OptionSelectionService<T>,
+    @Inject(IF_ACTIVE_ID) public id: number,
+    private ariaService: AriaService,
+    private el: ElementRef,
+    public commonStrings: ClrCommonStringsService,
+    private focusHandler: ComboboxFocusHandler<T>,
+    private toggleService: ClrPopoverToggleService,
     @Optional()
     @Inject(POPOVER_HOST_ANCHOR)
-    parentHost: ElementRef
+    parentHost: ElementRef,
+    @Inject(DOCUMENT) private document: any
   ) {
     if (!parentHost) {
       throw new Error('clr-options should only be used inside of a clr-combobox');
     }
-    super(injector, parentHost);
 
-    // Configure Popover
-    this.initializeSubscriptions();
-    this.configurePopover();
+    if (!this.optionsId) {
+      this.optionsId = 'clr-options-' + nbOptionsComponents++;
+    }
+  }
+
+  public searchText(input: string) {
+    return this.commonStrings.parse(this.commonStrings.keys.searching, { INPUT: input });
   }
 
   /**
-   * Configure Popover Direction and Close indicators
+   * Tests if the list of options is empty, meaning it doesn't contain any items
    */
-  private configurePopover(): void {
-    this.anchorPoint = Point.BOTTOM_LEFT;
-    this.popoverPoint = Point.LEFT_TOP;
-    this.closeOnOutsideClick = true;
+  public get emptyOptions() {
+    return !this.optionSelectionService.loading && this.items.length === 0;
   }
 
-  private initializeSubscriptions(): void {
-    // @TODO COMBOBOX: fix on combobox merge
-    // this.sub = this.toggleService.ignoredElementChange.subscribe((el: ElementRef) => {
-    //   if (el) {
-    //     this.ignoredElement = el.nativeElement;
-    //   }
-    //   this.sub.unsubscribe();
-    // });
+  @Input('id')
+  set optionsId(id: string) {
+    this.ariaService.ariaControls = id;
+    this.ariaService.ariaOwns = id;
+  }
+
+  get optionsId(): string {
+    return this.ariaService.ariaControls;
+  }
+
+  _items: QueryList<ClrOption<T>>;
+  @ContentChildren(ClrOption)
+  set items(items: QueryList<ClrOption<T>>) {
+    this._items = items;
+    this.focusHandler.addOptionValues(this._items.map(option => option.optionProxy));
+  }
+
+  get items(): QueryList<ClrOption<T>> {
+    return this._items;
+  }
+
+  loadingStateChange(state: ClrLoadingState): void {
+    this.loading = state === ClrLoadingState.LOADING;
+  }
+
+  ngAfterViewInit() {
+    this.focusHandler.listbox = this.el.nativeElement;
```

Comment:
Need a platform check

Reply:
Not yet. It's just assignment of a nativeElement. We only apply any function on it at `ComboboxFocusHandler.addFocusOnBlurListener()` and there we have the platform check.
We are transferring nativeElements without platform check at many places in the code.

Reply:
I'm more worried about the subscription below, though it should never fire a scroll event in SSR.